src/scripts/main.js

```diff
@@ -1,3 +1,271 @@
+    const tr = document.createElement('tr');
+
+    for (const value of Object.values(row)) {
+      const td = document.createElement('td');
+
+      td.textContent = value;
+      tr.append(td);
+    }
+
+    tbody.append(tr);
+  });
+}
+
+head.addEventListener('click', (e) => {
+  const column = e.target.closest('th').textContent.trim();
+
+  switch (column) {
+    case 'Name':
+    case 'Position':
+    case 'Office':
+      sortData(column);
+      break;
+    case 'Age':
+      sortData(column, true);
+      break;
+    case 'Salary':
+      sortData(column, true);
+      break;
+  }
+
+  createNewTable(sortedRows);
+});
+
+tbody.addEventListener('click', (e) => {
+  const target = e.target.closest('tr');
+
+  [...tbody.children].forEach((tr) => {
+    tr.classList.remove('active');
+  });
+
+  target.classList.add('active');
+});
+
+const form = document.createElement('form');
+
+form.className = 'new-employee-form';
+
+function createLabeledInput(labelText, inputType, namePeson, dataQa) {
+  const label = document.createElement('label');
+
+  label.textContent = labelText;
+
+  const input = document.createElement('input');
+
+  input.type = inputType;
+  input.name = namePeson;
+  input.dataset.qa = dataQa;
+
+  label.append(input);
+
+  return label;
+}
+
+const nameInput = createLabeledInput('Name:', 'text', 'name', 'name');
+const positionInput = createLabeledInput(
+  'Position:',
+  'text',
+  'position',
+  'position',
+);
+const ageInput = createLabeledInput('Age:', 'number', 'age', 'age');
+const salaryInput = createLabeledInput('Salary:', 'number', 'salary', 'salary');
+
+const officeLabel = document.createElement('label');
+const officeSelect = document.createElement('select');
+
+officeLabel.textContent = 'Office:';
+officeSelect.name = 'office';
+officeSelect.dataset.qa = 'office';
+officeSelect.required = true;
+
+const offices = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+offices.forEach((office) => {
+  const option = document.createElement('option');
+
+  option.value = office;
+  option.textContent = office;
+  officeSelect.append(option);
+});
+officeLabel.append(officeSelect);
+
+const saveButton = document.createElement('button');
+
+saveButton.type = 'submit';
+saveButton.textContent = 'Save to table';
+
+form.append(nameInput);
+form.append(positionInput);
+form.append(officeLabel);
+form.append(ageInput);
+form.append(salaryInput);
+form.append(saveButton);
+
+document.body.append(form);
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const formData = new FormData(form);
+
+  const newPerson = {
+    Name: formData.get('name'),
+    Position: formData.get('position'),
+    Office: formData.get('office'),
+    Age: formData.get('age'),
+    Salary: `$${Number(formData.get('salary')).toLocaleString()}`,
+  };
+
+  const nameIsValid = newPerson.Name.length >= 4;
+  const ageIsValid = newPerson.Age >= 18 && newPerson.Age <= 90;
```

Comment:
The validation logic for `nameIsValid` and `ageIsValid` is correct, but ensure that all necessary fields are validated before proceeding with form submission.